scheduler/src/cook/mesos/task.clj

```diff
@@ -77,10 +77,27 @@
             (assoc "EXECUTOR_PROGRESS_OUTPUT_FILE_ENV" "EXECUTOR_PROGRESS_OUTPUT_FILE_NAME"
                    "EXECUTOR_PROGRESS_OUTPUT_FILE_NAME" progress-output-file))))
 
+(defn merge-container-defaults
+  "Takes a job container specification and applies any defaults from the config"
+  [container]
+  (when container
+    (let [{:keys [volumes]} (config/container-defaults)
+          get-path (fn [volume] (or (:container-path volume)
+                                    (:host-path volume)))]
+      (cond-> container
+              volumes
+              (update :volumes (fn [container-volumes]
+                                 (let [volumes-to-add (filter (fn [default-volume]
+                                                              (not-any? (fn [container-volume]
+                                                                          (.startsWith (get-path default-volume)
+                                                                                       (get-path container-volume))) container-volumes))
```

Comment:
Nitpick, would have preferred the second argument to the `filter` (i.e. `container-volumes`) in a new line as the filter function spans over multiple lines.`